### PR TITLE
fix: editor can add dependencies in default project

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Editors should be able to add dependencies in a default project.

Details:
* adding new permission for editor
* editor is selection is dynamic by the role name instead of hardcoding 2 which is magical
* making sure the queries are idempotent and can be run multiple times

### CODE REVIEW

File: src/migrations/20240904084114-add-update-feature-dependency-editor.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+exports.up = function (db, cb) {
+    db.runSql(
+        `
+        INSERT INTO role_permission (role_id, permission)
+        SELECT id, 'UPDATE_FEATURE_DEPENDENCY'
+        FROM roles WHERE name = 'Editor'
+        AND EXISTS (SELECT 1 FROM roles WHERE name = 'Editor')
```
Comment (kwasniew) [inline src/migrations/20240904084114-add-update-feature-dependency-editor.js:7]: prevent a case where someone manually deleted editor

File: src/migrations/20240904084114-add-update-feature-dependency-editor.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        AND NOT EXISTS (
+            SELECT 1
```
Comment (kwasniew) [inline src/migrations/20240904084114-add-update-feature-dependency-editor.js:9]: prevent double insert

**Review by sjaanus — CHANGES_REQUESTED** [review]
Verify that the role is rool role.

**Review by sjaanus — APPROVED** [review]
Query itself looks good.

**Pull request merged**